modules/@apostrophecms/page/index.js

```diff
@@ -161,6 +161,35 @@ module.exports = {
             };
           }
 
+          if (type.length) {
+            const manager = self.apos.doc.getManager(type);
+            if (!manager) {
+              throw self.apos.error('invalid');
+            }
+
+            const query = self.getRestQuery(req);
+            query
+              .type(type)
+              .ancestors(false)
+              .children(false)
+              .attachments(false)
+              .perPage(manager.options.perPage);
+
+            // populates totalPages when perPage is present
+            await query.toCount();
+
+            const docs = await query.toArray();
+
+            return {
+              results: docs.map(doc => manager.removeForbiddenFields(req, doc)),
+              pages: query.get('totalPages'),
+              currentPage: query.get('page') || 1,
+              ...(query.get('choicesResults') && {
+                choices: query.get('choicesResults')
+              })
```

Comment:
🤔 `choices` is always undefined here, but it is the same for pieces...

Reply:
If `req.query.choices` is an array of field names then I believe this will get populated. Just FYI.